### PR TITLE
Add parent call spacing sniff

### PR DESCRIPTION
**Pull request by grongor**
Fixes https://github.com/slevomat/coding-standard/issues/678

I'm not sure if we should create a separate sniff extending the abstract class, or just keep it here. What do you think? Splitting is trivial ... just pitch me a good name :D

### CODE REVIEW

**Comment by kukulich** [comment]
I definitely want new sniff :) It should be in `Classes` directory.

We probably  should support `self::call()` too.

And more tests should be needed for codes like:

```
foreach (self::getAvailableEnums() as $interval) {
```

**Comment by grongor** [comment]
So put it in Classes directory and extend from here?

Also, don't think that the `self::call()` makes much sense, as it's just a generic method call ... if you want to support that, then we should also support regular methods. Yeah, it doesn't make sense to me.

I was a bit on edge about `static::call()` too, but I think it's "special enough" and more like `parent::call()`.

Anyway, what would be the sniff name? :)

**Comment by kukulich** [comment]
I think we can create `Classes.ParentCallSpacingSniff` and support only `parent`.

The `static::call()` is strange case for me. If you want to support it too I would add second sniff `Classes.StaticCallSpacingSniff`

**Comment by grongor** [comment]
It's not very much used, but yeah I consider it a kind of special case...but yeah, I talked with my colleague and we decided that it's not worth to have a sniff for it :D I'll post the parent sniff soon.

File: tests/Sniffs/Classes/ParentCallSpacingSniffTest.php [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+		self::assertNoSniffErrorInFile($report);
+	}
+
+	public function testErrors(): void
+	{
+		$report = self::checkFile(__DIR__ . '/data/parentCallSpacingErrors.php');
+
+		self::assertSame(2, $report->getErrorCount());
+
+		self::assertSniffError($report, 5, AbstractControlStructureSpacingSniff::CODE_INCORRECT_LINES_COUNT_AFTER_CONTROL_STRUCTURE, 'Expected 1 lines after "parent", found 0.');
```
Comment (kukulich) [inline tests/Sniffs/Classes/ParentCallSpacingSniffTest.php]: Use `ParentCallSpacingSniff::`

Comment (grongor) [reply tests/Sniffs/Classes/ParentCallSpacingSniffTest.php]: yeah ofc

**Comment by grongor** [comment]
> 2. I don't know how to disable useless settings `tokensToCheck` for this sniff :(

Maybe I just fill the property with this sniff, so it's there automatically? ... you kind off broke this in here https://github.com/slevomat/coding-standard/commit/307fcc2c418d24d7c776e7681278b8cb2e8d7f39 :D

**Comment by kukulich** [comment]
We need to move the property `$tokensToCheck` to `BlockControlStructureSpacingSniff` and `JumpStatementsSpacingSniff`.

**Comment by grongor** [comment]
Done

**Pull request merged**